packages/myst-parser/src/myst.ts

```diff
@@ -93,7 +93,9 @@ export function mystParse(content: string, opts?: Options) {
   const parsedOpts = parseOptions(opts);
   const tokenizer = createTokenizer(parsedOpts);
   const tree = tokensToMyst(content, tokenizer.parse(content, { vfile }), parsedOpts.mdast);
-  applyDirectives(tree, parsedOpts.directives, parsedOpts.vfile);
+  applyDirectives(tree, parsedOpts.directives, parsedOpts.vfile, {
+    parseMyST: (source: string) => mystParse(source, opts),
+  });
   applyRoles(tree, parsedOpts.roles, parsedOpts.vfile);
```

Comment:
Do roles get the ctx as well?

Reply:
Maybe not based on your comments above, we can always add it if we need to later.